rust/connlib/tunnel/src/peer.rs

```diff
@@ -356,7 +360,14 @@ impl ClientOnGateway {
     // in case that 2 or more resources have overlapping rules.
     fn recalculate_filters(&mut self) {
         self.filters = IpNetworkTable::new();
-        for resource in self.resources.values() {
+        self.recalculate_filters();
```

Comment:
```suggestion
        self.recalculate_cidr_filters();
```

I assume you don't want to recurse here :)

Reply:
fixed! :p I was extremly tired when I did that yesterday hah